expect-expect: explicit assertFunctionNames with documented defaults misses chained .expect() calls that the implicit default catches

## Summary

[The docs for `vitest/expect-expect`](https://oxc.rs/docs/guide/usage/linter/rules/vitest/expect-expect.html) state that the default `assertFunctionNames` list is `["expect", "expectTypeOf", "assert", "assertType"]`. Supplying that same list explicitly in the config changes the rule's matching behavior as chained method-call assertions like `supertest`'s `await request(app).post('/').send(body).expect(501)` are flagged as "Test has no assertions", even though the rule with no override (i.e. relying on defaults) correctly recognizes them as assertions.

## Reproduction

A single test file that uses supertest's chained `.expect()`:

```
// repro.test.ts
import express from 'express'
import supertest from 'supertest'

const app = express()
app.use(express.json())
app.post('/', (_, res) => res.status(501).send())

describe('supertest chained expect', () => {
  it('returns 501 for the request body', async () => {
    await supertest(app).post('/').send({ foo: 'bar' }).expect(501)
  })
})
```

### Case A - implicit defaults (passes ✅)

```
// oxlint.config.ts
export default defineConfig({
  plugins: ['vitest'],
  rules: {
    'vitest/expect-expect': 'error'
  },
})
```

`oxlint repro.test.ts` → 0 errors

### Case B - explicit defaults exactly as documented (fails ❌)

```
// oxlint.config.ts (excerpt)
export default defineConfig({
  plugins: ['vitest'],
  rules: {
    'vitest/expect-expect': [
      'error',
      { assertFunctionNames: ['expect', 'expectTypeOf', 'assert', 'assertType'] },
    ],
  },
})
```

oxlint repro.test.ts → reports:

repro.test.ts:9:3: error vitest(expect-expect): Test has no assertions
  help: Add assertion(s) in this Test

## Expected

Setting `assertFunctionNames` to the documented default list should be a no-op, the rule should report the same diagnostics as with no override.

## Actual

Providing any explicit `assertFunctionNames` value (including the documented default) switches the rule to a stricter matching algorithm that no longer treats `*.expect()` deeper than one chain segment as an assertion. 

## Impact

Users who want to extend the assertion list have no documented way to do so without simultaneously changing detection on chained `.expect()` calls.

--- a/crates/oxc_linter/src/rules/jest/expect_expect.rs
+++ b/crates/oxc_linter/src/rules/jest/expect_expect.rs
@@ -235,6 +235,11 @@ fn test() {
             Some(serde_json::json!([{ "assertFunctionNames": ["request.foo**.expect"] }])),
         ),
         (
+            "test('should fail with implicit defaults', () => request.post().send().expect(457));",
+            None,
+        ),
+        (r#"it("should fail with implicit defaults",() => expectSaga(mySaga).returns());"#, None),
+        (
             "test('should fail', () => tester.request(123));",
             Some(serde_json::json!([{ "assertFunctionNames": ["request.*"] }])),
         ),

--- a/crates/oxc_linter/src/rules/shared/jest_vitest/expect_expect.rs
+++ b/crates/oxc_linter/src/rules/shared/jest_vitest/expect_expect.rs
@@ -74,12 +74,19 @@ impl Default for ExpectExpectConfig {
     }
 }
 
+fn default_assert_function_names_jest() -> Vec<CompactStr> {
+    vec!["expect".into()]
+}
+
+fn default_assert_function_names_vitest() -> Vec<CompactStr> {
+    vec!["expect".into(), "expectTypeOf".into(), "assert".into(), "assertType".into()]
+}
+
 impl ExpectExpectConfig {
     #[expect(clippy::unnecessary_wraps)]
     pub fn from_configuration(value: &serde_json::Value) -> Result<Self, serde_json::error::Error> {
-        let default_assert_function_names_jest = vec!["expect".into()];
-        let default_assert_function_names_vitest =
-            vec!["expect".into(), "expectTypeOf".into(), "assert".into(), "assertType".into()];
+        let default_assert_function_names_jest = default_assert_function_names_jest();
+        let default_assert_function_names_vitest = default_assert_function_names_vitest();
         let config = value.get(0);
 
         let assert_function_names = config
@@ -88,7 +95,7 @@ impl ExpectExpectConfig {
             .map(|v| {
                 v.iter()
                     .filter_map(serde_json::Value::as_str)
-                    .map(convert_pattern)
+                    .map(CompactStr::from)
                     .collect::<Vec<_>>()
             });
 
@@ -151,8 +158,10 @@ fn run<'a>(
             } else {
                 &rule.assert_function_names_jest
             };
+            let assert_function_names =
+                assert_function_names.iter().map(|name| convert_pattern(name)).collect::<Vec<_>>();
 
-            let mut visitor = AssertionVisitor::new(ctx, assert_function_names);
+            let mut visitor = AssertionVisitor::new(ctx, &assert_function_names);
 
             // Visit each argument of the test call
             for argument in &call_expr.arguments {

--- a/crates/oxc_linter/src/rules/vitest/expect_expect.rs
+++ b/crates/oxc_linter/src/rules/vitest/expect_expect.rs
@@ -302,6 +302,16 @@ fn test() {
             Some(serde_json::json!([{ "assertFunctionNames": ["request.foo**.expect"] }])),
         ),
         (
+            "test('should fail with implicit defaults', () => request.post().send().expect(457));",
+            None,
+        ),
+        (
+            "test('should fail with explicit defaults', () => request.put().send().expect(458));",
+            Some(serde_json::json!([{
+                "assertFunctionNames": ["expect", "expectTypeOf", "assert", "assertType"],
+            }])),
+        ),
+        (
             "test('should fail', () => tester.request(123));",
             Some(serde_json::json!([{ "assertFunctionNames": ["request.*"] }])),
         ),

